File: libraries/DemoApp.Domain/Paging/Models/TableMetadata.cs
namespace DemoApp.Domain.Paging.Models
{
    public class TableMetadata
    {
        public List<FilterMetadata>? FilterMetadata { get; set; }
        public List<SortMetadata>? SortMetadata { get; set; }
        public PagingMetadata? PagingMetadata { get; set; }
    }
}

File: DemoApp.spa/src/app/shared/entities/models/table-metadata.ts
import { FilterMetadata } from './filter-metadata';
import { PagingMetadata } from './paging-metadata';
import { SortMetadata } from './sort-metadata';

export class TableMetadata{
    filterMetadata: FilterMetadata;
    sortMetadata: SortMetadata;
    pagingMetadata: PagingMetadata;
}
